### ISSUE TITLE
[RyuJIT/ARM32] Assertion failed '((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume)'

### ISSUE THREAD

**Issue by hqueue** [dotnet/coreclr#13565]
Following assertion is observed from `JIT/Methodical/divrem/rem/u8rem_cs_ro/` with `COMPlus_AltJit=*`

```
               Assert failure(PID 2984 [0x00000ba8], Thread: 2984 [0x0ba8]): Assertion failed '((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume)' in 'u8rem:Main():int' (IL size 27711)

                   File: /opt/code/github/hqueue/coreclr/src/jit/lsra.cpp Line: 3638

```

Same assertion observed from following tests:
```
FAILED   - [   2][   6s]JIT/Methodical/MDArray/DataTypes/decimal_cs_d/decimal_cs_d.sh
FAILED   - [   3][   6s]JIT/Methodical/MDArray/DataTypes/decimal_cs_do/decimal_cs_do.sh
FAILED   - [   4][   7s]JIT/Methodical/MDArray/DataTypes/decimal_cs_r/decimal_cs_r.sh
FAILED   - [   5][   7s]JIT/Methodical/MDArray/DataTypes/decimal_cs_ro/decimal_cs_ro.sh
FAILED   - [   6][   5s]JIT/Methodical/divrem/div/decimaldiv_cs_d/decimaldiv_cs_d.sh
FAILED   - [   7][   6s]JIT/Methodical/divrem/div/decimaldiv_cs_do/decimaldiv_cs_do.sh
FAILED   - [   8][   6s]JIT/Methodical/divrem/div/decimaldiv_cs_r/decimaldiv_cs_r.sh
FAILED   - [   9][   7s]JIT/Methodical/divrem/div/decimaldiv_cs_ro/decimaldiv_cs_ro.sh
FAILED   - [  10][   6s]JIT/Methodical/divrem/div/i4div_cs_d/i4div_cs_d.sh
FAILED   - [  11][   6s]JIT/Methodical/divrem/div/i4div_cs_do/i4div_cs_do.sh
FAILED   - [  12][   6s]JIT/Methodical/divrem/div/i4div_cs_r/i4div_cs_r.sh
FAILED   - [  13][   6s]JIT/Methodical/divrem/div/i4div_cs_ro/i4div_cs_ro.sh
FAILED   - [  14][   6s]JIT/Methodical/divrem/div/i8div_cs_d/i8div_cs_d.sh
FAILED   - [  15][   6s]JIT/Methodical/divrem/div/i8div_cs_do/i8div_cs_do.sh
FAILED   - [  16][   5s]JIT/Methodical/divrem/div/i8div_cs_r/i8div_cs_r.sh
FAILED   - [  17][   5s]JIT/Methodical/divrem/div/i8div_cs_ro/i8div_cs_ro.sh
FAILED   - [  18][   6s]JIT/Methodical/divrem/div/u4div_cs_d/u4div_cs_d.sh
FAILED   - [  19][   6s]JIT/Methodical/divrem/div/u4div_cs_do/u4div_cs_do.sh
FAILED   - [  20][   6s]JIT/Methodical/divrem/div/u4div_cs_r/u4div_cs_r.sh
FAILED   - [  21][   6s]JIT/Methodical/divrem/div/u4div_cs_ro/u4div_cs_ro.sh
FAILED   - [  22][   6s]JIT/Methodical/divrem/div/u8div_cs_d/u8div_cs_d.sh
FAILED   - [  23][   6s]JIT/Methodical/divrem/div/u8div_cs_do/u8div_cs_do.sh
FAILED   - [  24][   6s]JIT/Methodical/divrem/div/u8div_cs_r/u8div_cs_r.sh
FAILED   - [  25][   6s]JIT/Methodical/divrem/div/u8div_cs_ro/u8div_cs_ro.sh
FAILED   - [  26][   6s]JIT/Methodical/divrem/rem/decimalrem_cs_d/decimalrem_cs_d.sh
FAILED   - [  27][   7s]JIT/Methodical/divrem/rem/decimalrem_cs_do/decimalrem_cs_do.sh
FAILED   - [  28][   6s]JIT/Methodical/divrem/rem/decimalrem_cs_r/decimalrem_cs_r.sh
FAILED   - [  29][   6s]JIT/Methodical/divrem/rem/decimalrem_cs_ro/decimalrem_cs_ro.sh
FAILED   - [  30][   6s]JIT/Methodical/divrem/rem/i4rem_cs_d/i4rem_cs_d.sh
FAILED   - [  31][   6s]JIT/Methodical/divrem/rem/i4rem_cs_do/i4rem_cs_do.sh
FAILED   - [  32][   6s]JIT/Methodical/divrem/rem/i4rem_cs_r/i4rem_cs_r.sh
FAILED   - [  33][   6s]JIT/Methodical/divrem/rem/i4rem_cs_ro/i4rem_cs_ro.sh
FAILED   - [  34][   6s]JIT/Methodical/divrem/rem/i8rem_cs_d/i8rem_cs_d.sh
FAILED   - [  35][   6s]JIT/Methodical/divrem/rem/i8rem_cs_do/i8rem_cs_do.sh
FAILED   - [  36][   6s]JIT/Methodical/divrem/rem/i8rem_cs_r/i8rem_cs_r.sh
FAILED   - [  37][   7s]JIT/Methodical/divrem/rem/i8rem_cs_ro/i8rem_cs_ro.sh
FAILED   - [  38][   6s]JIT/Methodical/divrem/rem/u4rem_cs_d/u4rem_cs_d.sh
FAILED   - [  39][   6s]JIT/Methodical/divrem/rem/u4rem_cs_do/u4rem_cs_do.sh
FAILED   - [  40][   6s]JIT/Methodical/divrem/rem/u4rem_cs_r/u4rem_cs_r.sh
FAILED   - [  41][   6s]JIT/Methodical/divrem/rem/u4rem_cs_ro/u4rem_cs_ro.sh
FAILED   - [  42][   5s]JIT/Methodical/divrem/rem/u8rem_cs_d/u8rem_cs_d.sh
FAILED   - [  43][   5s]JIT/Methodical/divrem/rem/u8rem_cs_do/u8rem_cs_do.sh
FAILED   - [  44][   5s]JIT/Methodical/divrem/rem/u8rem_cs_r/u8rem_cs_r.sh
FAILED   - [  45][   4s]JIT/Methodical/divrem/rem/u8rem_cs_ro/u8rem_cs_ro.sh
```

**Comment by hqueue** [comment]
It seems that this is a following issue of PR #13198 and #13470.

@BruceForstall Please assign this to me.

**Comment by sergign60** [comment]
@hqueue I've got the same issue when I've tried to compile HelloWorld appiication with CoreRT 

```
Compiling System.Collections.Generic.ArraySortHelper`1[Relocation][ILCompiler.DependencyAnalysis.Relocation]:SwapIfGreater(ref,ref,int,int)

Assert failure(PID 18320 [0x00004790], Thread: 18320 [0x4790]): Assertion failed '((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume)' in 'System.Collections.Generic.ArraySortHelper`1[Relocation][ILCompiler.DependencyAnalysis.Relocation]:SwapIfGreater(ref,ref,int,int)' (IL size 74)

    File: /share/CORERT/coreclr/src/jit/lsra.cpp Line: 3638
    Image: /opt/usr/home/owner/arm_target_dir/corerun

```

**Comment by CarolEidt** [comment]
Fixed with #13579
